classes/assets/CssMinifier.php

```diff
@@ -28,7 +28,13 @@
 
 class CssMinifierCore
 {
-    public static function minify(array $files, $destination)
+    /**
+     * @param string[] $files
+     * @param string $destination
+     *
+     * @return string Minified data
+     */
+    public static function minify(array $files, string $destination)
```

Comment:
same here, it's a BC, could have been an object with `__toString`

Reply:
You mean that an object, when `foreach` is applied to it, if it has `__toString()` function, this will return `__toString()` result and it will loop over it ?

... seriously 😭 😅 ?

Reply:
I don't know, but as @eternoendless said, changing the signature is a BC :/

Reply:
Have a lot of phpdoc update in stock on my side.
Modules code analysis with phpstan return to many error due to bad phpdoc
Prefer to not make code changes to avoid tests from QA.
So what to do ?
Not change type of parameters and returns and add comment to explain expected type ?

Reply:
@Matt75 We should probably ignore the issues from the past using [PHPStan baseline](https://medium.com/@ondrejmirtes/phpstans-baseline-feature-lets-you-hold-new-code-to-a-higher-standard-e77d815a5dff). Only new code should be 100% bugfree according to phpstan.

Because legacy code is too big, too messy and sometimes (like here) cannot be updated because of SemVer 😭

Reply:
Ok but phpdoc is useful for all developers have to interact with legacy classes, so I would like to share my personal work on it.
I just want to know what to do for type not matches signature, should I use mixed and add a comment to explain expected type ?
I made a PR for Product class but I have a lot of PR in stock ^^

Reply:
> should I use mixed and add a comment to explain expected type ?
I think this is the best solution. We are sure there is no BC break and you can explain what content is "most of the time".

Reply:
@PierreRambaud BC break fixed in https://github.com/PrestaShop/PrestaShop/pull/18185/commits/e495740557e289cac08d92f6319badd4233dbd63